aws/ec2metadata/api.go

```diff
@@ -65,19 +93,57 @@ func (c *EC2Metadata) GetDynamicData(p string) (string, error) {
 	return output.Content, req.Send()
 }
 
+// Parses a PEM-encoded certificate
+func parseCertificate(certPem string) (*x509.Certificate, error) {
+	dec, _ := pem.Decode([]byte(certPem))
+	if dec == nil {
+		return nil, fmt.Errorf("failed to find any PEM-encoded data block")
+	}
+	return x509.ParseCertificate(dec.Bytes)
+}
+
 // GetInstanceIdentityDocument retrieves an identity document describing an
-// instance. Error is returned if the request fails or is unable to parse
-// the response.
-func (c *EC2Metadata) GetInstanceIdentityDocument() (EC2InstanceIdentityDocument, error) {
-	resp, err := c.GetDynamicData("instance-identity/document")
+// instance and verifies its signature with provided certificate.
+// Error is returned if failed to parse the certificate, or the request fails,
+// or is unable to parse the response, or the signature verification fails.
+func (c *EC2Metadata) GetInstanceIdentityDocument(certPem string) (EC2InstanceIdentityDocument, error) {
```

Comment:
Changing the signature of this function would be a breaking change.  I think this change also removes requesting the identity document as well.